activerecord/lib/active_record/attribute_set/yaml_encoder.rb

```diff
@@ -9,7 +9,7 @@ def initialize(default_types)
 
       def encode(attribute_set, coder)
         coder['concise_attributes'] = attribute_set.each_value.map do |attr|
-          if attr.type.equal?(default_types[attr.name])
+          if attr.type.equal?(default_types[attr.name]) && !attr.changed?
```

Comment:
What does this fix? Why does it fix it? This change feels unrelated to the intent of the existing code. Can you give more context as to why this fixes the root problem and isn't a bandaid?